src/main/java/org/embulk/input/athena/AthenaInputPlugin.java

```diff
@@ -281,10 +327,76 @@ protected Connection getAthenaConnection(PluginTask task) throws ClassNotFoundEx
         return DriverManager.getConnection(task.getAthenaUrl(), properties);
     }
 
+
+    private ColumnGetterFactory newColumnGetterFactory(PageBuilder pageBuilder, ZoneId dateTimeZone)
+    {
+        return new AthenaColumnGetterFactory(pageBuilder, dateTimeZone);
+    }
+
+    private static class AthenaColumnGetterFactory extends ColumnGetterFactory {
+        public AthenaColumnGetterFactory(PageBuilder to, ZoneId defaultTimeZone) {
+            super(to, defaultTimeZone);
+        }
+
+        @Override
+        protected String sqlTypeToValueType(JdbcColumn column, int sqlType) {
+            try {
+                return super.sqlTypeToValueType(column, sqlType);
+            } catch (UnsupportedOperationException e) {
```

Comment:
There is a reference in the parent to 'select:' option that is not present in this plugin.

However, since the implementation feels a little strange,
it may be better to reimplement the necessary parts rather than using extends ColumnGetterFactory,
although it will require more changes.

Reply:
> There is a reference in the parent to 'select:' option that is not present in this plugin.

sorry, I had missed it.
It's fine as is 👍